[rdf] gcc14 issue warning in `RDF/InterfaceUtils.hxx`

### Explain what you would like to see improved and how.

One should fix following warning.

Appears in following line:  https://github.com/root-project/root/blob/master/tree/dataframe/inc/ROOT/RDF/InterfaceUtils.hxx#L270

```
In file included from /usr/include/c++/14/string:51,
                 from /home/linev/git/webgui/core/meta/inc/TSchemaHelper.h:17,
                 from /home/linev/git/webgui/core/meta/inc/TGenericClassInfo.h:21,
                 from /home/linev/git/webgui/core/base/inc/Rtypes.h:193,
                 from /home/linev/git/webgui/core/base/inc/TObject.h:17,
                 from /home/linev/git/webgui/core/base/inc/TNamed.h:25,
                 from /home/linev/git/webgui/core/base/inc/TDirectory.h:24,
                 from /home/linev/git/webgui/core/base/inc/TROOT.h:28,
                 from /home/linev/git/webgui/tree/dataframe/inc/ROOT/RDataFrame.hxx:19,
                 from /home/linev/git/webgui/tree/dataframe/test/dataframe_utils.cxx:1:
In static member function ‘static constexpr _Up* std::__copy_move<_IsMove, true, std::random_access_iterator_tag>::__copy_m(_Tp*, _Tp*, _Up*) [with _Tp = long unsigned int; _Up = long unsigned int; bool _IsMove = false]’,
    inlined from ‘constexpr _OI std::__copy_move_a2(_II, _II, _OI) [with bool _IsMove = false; _II = long unsigned int*; _OI = long unsigned int*]’ at /usr/include/c++/14/bits/stl_algobase.h:521:30,
    inlined from ‘constexpr _OI std::__copy_move_a1(_II, _II, _OI) [with bool _IsMove = false; _II = long unsigned int*; _OI = long unsigned int*]’ at /usr/include/c++/14/bits/stl_algobase.h:548:42,
    inlined from ‘constexpr _OI std::__copy_move_a(_II, _II, _OI) [with bool _IsMove = false; _II = long unsigned int*; _OI = long unsigned int*]’ at /usr/include/c++/14/bits/stl_algobase.h:555:31,
    inlined from ‘constexpr _OI std::copy(_II, _II, _OI) [with _II = long unsigned int*; _OI = long unsigned int*]’ at /usr/include/c++/14/bits/stl_algobase.h:651:7,
    inlined from ‘constexpr std::vector<bool, _Alloc>::iterator std::vector<bool, _Alloc>::_M_copy_aligned(const_iterator, const_iterator, iterator) [with _Alloc = std::allocator<bool>]’ at /usr/include/c++/14/bits/stl_bvector.h:1342:28,
    inlined from ‘constexpr void std::vector<bool, _Alloc>::_M_reallocate(size_type) [with _Alloc = std::allocator<bool>]’ at /usr/include/c++/14/bits/vector.tcc:1059:40,
    inlined from ‘constexpr void std::vector<bool, _Alloc>::reserve(size_type) [with _Alloc = std::allocator<bool>]’ at /usr/include/c++/14/bits/stl_bvector.h:1130:17,
    inlined from ‘ROOT::Internal::RDF::BuildAction<char, unsigned char, int, unsigned int, short int, short unsigned int, double, float, long long int, long long unsigned int, bool, ROOT::Detail::RDF::RLoopManager>(const ROOT::RDF::ColumnNames_t&, const 
std::shared_ptr<SnapshotHelperArgs>&, unsigned int, std::shared_ptr<ROOT::Detail::RDF::RLoopManager>, ActionTags::Snapshot, const RColumnRegister&)::<lambda()>’ at tree/dataframe/inc/ROOT/RDF/InterfaceUtils.hxx:270:20:
/usr/include/c++/14/bits/stl_algobase.h:452:30: warning: ‘void* __builtin_memmove(void*, const void*, long unsigned int)’ forming offset 8 is out of the bounds [0, 8] [-Warray-bounds=]
  452 |             __builtin_memmove(__result, __first, sizeof(_Tp) * _Num);
      |             ~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```



### ROOT version

master

### Installation method

build from source

### Operating system

OpenSUSE linux

### Additional context

_No response_

--- a/tree/dataframe/inc/ROOT/RDF/InterfaceUtils.hxx
+++ b/tree/dataframe/inc/ROOT/RDF/InterfaceUtils.hxx
@@ -266,10 +266,10 @@ BuildAction(const ColumnNames_t &colNames, const std::shared_ptr<SnapshotHelperA
    const auto &options = snapHelperArgs->fOptions;
 
    auto makeIsDefine = [&] {
-      std::vector<bool> isDef;
-      isDef.reserve(sizeof...(ColTypes));
-      for (auto i = 0u; i < sizeof...(ColTypes); ++i)
-         isDef.push_back(colRegister.IsDefineOrAlias(colNames[i]));
+      auto sz = sizeof...(ColTypes);
+      std::vector<bool> isDef(sz);
+      for (auto i = 0u; i < sz; ++i)
+         isDef[i] = colRegister.IsDefineOrAlias(colNames[i]);
       return isDef;
    };
    std::vector<bool> isDefine = makeIsDefine();

